src/components/HeatmapCell.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const TableCell = styled.td`
+  text-align: center;
+  width: 39px;
+  height: 39px;
+  padding: 0;
+  color: white;
+  font-weight: bold;
+  font-size: ${({ theme }) => theme.font.size.small};
+  background-color: ${({ theme, colorValue }) => theme.color.count[colorValue]};
+  border: 1px solid ${({ theme, colorValue, clicked }) => (
+    clicked ? theme.color.tablerowheader : theme.color.count[colorValue]
+  )};
+  &:hover {
+    border: 1px solid ${({ theme }) => theme.color.tablerowheader};
+  }
+`;
+
+const HeatmapCell = ({
+  children, colorValue, handleCellClick, id, clickedCellId,
+}) => {
+  const [clicked, setClicked] = useState(false);
+
+  const handleClick = () => {
+    setClicked(true);
+    handleCellClick(id);
+  };
+
+  useEffect(() => {
+    if (clickedCellId === id) {
+      setClicked(true);
+    } else {
+      setClicked(false);
+    }
+  }, [clickedCellId, id]);
```

Comment:
This is a very common mistake and you might be surprised how much simpler it can be :smile:

`clicked` is just derived from the `clickedCellId` and `id` props`. So you can simply replace the state variable by `const clicked = clickedCellId === id;` as you did in the `useEffect`. Try it out, your component will look much simpler

Reply:
woah! So much better.  Thanks!  I was overthinking it, it seems